src/protocols/core.rs

```diff
@@ -415,48 +421,48 @@ unsafe fn handle_pvalidate_request(vmsa: *mut Vmsa) {
         return;
     }
 
-    let va: VirtAddr = match pgtable_map_pages_private(gpa, CAA_MAP_SIZE) {
-        Ok(v) => v,
-        Err(_e) => return,
-    };
-
-    let request: *mut PvalidateRequest = va.as_mut_ptr();
-    if (*request).entries() == 0 || (*request).entries() < (*request).next() {
-        return;
-    }
-
-    // Request data cannot cross a 4K boundary
-    let va_end: VirtAddr = va
-        + size_of::<PvalidateRequest>()
-        + ((*request).entries() as usize * size_of::<PvalidateEntry>())
-        - 1_u64;
-
-    if va.align_down(PAGE_SIZE) != va_end.align_down(PAGE_SIZE) {
-        return;
-    }
-
     let mut flush: bool = false;
-    let mut e_va: VirtAddr = va + size_of::<PvalidateRequest>();
-    while (*request).next() < (*request).entries() {
-        let entry: *const PvalidateEntry = e_va.as_ptr();
+    {
```

Comment:
Maybe I'm missing something, why do the braces need to be added?  I guess I don't understand why they're there and it makes the diff much harder to read.

Reply:
Maybe try to view the diff while ignoring whitespace (`git diff -w`, and there's also an option in the github UI) can help with the indent change.

It's because we need to unmap the page before the end of the function where we do:

```
    if flush {
        invlpgb_all();
        tlbsync();
    }
```

But now I think -- maybe I'll just call `drop(map);` ([doc](https://doc.rust-lang.org/std/mem/fn.drop.html)) in the original location of the unmap, instead of insisting on the indented scopes with braces.

Reply:
> But now I think -- maybe I'll just call `drop(map);` ([doc](https://doc.rust-lang.org/std/mem/fn.drop.html)) in the original location of the unmap, instead of insisting on the indented scopes with braces.

It doesn't look like you did this.

Reply:
Yes, I forgot about the pvalidate patch. I'll change it as well.